File: src/lib/weight.ts
// The client refreshes weight and weightMax independently and reports a max of 0 before it has been
// told, against which every weight in the game is overweight. A mining run ended at
// *overweight (436/453)* on exactly that. Every read of the limit goes through here.
export const overweight = (buffer = 0): boolean =>
  player.weightMax > 0 && player.weight > player.weightMax - buffer;

File: legion/src/uo/weight.py
from uo.entity import player


# Unknown is not overweight, the same call WeightMax == 0 gets: WeightMax reads 0 before the client
# has been told, against which every weight is overweight - a live mining run ended at 436/453 on it
def over_buffer(buffer):
    me = player()

    if me is None:
        return False

    ceiling = me.WeightMax

    return ceiling > 0 and me.Weight > ceiling - buffer


def too_heavy():
    return over_buffer(0)
